pkg/controller/controller_test.go

```diff
@@ -0,0 +1,48 @@
+package controller
+
+import (
+	"context"
+	"testing"
+	"time"
+)
+
+func TestController(t *testing.T) {
+	ctx, cncl := context.WithTimeout(context.Background(), 1*time.Second)
+	defer cncl()
+	ctx1, cncl1 := context.WithCancel(context.Background())
```

Comment:
Variable names could be improve.  may be something like `ctxWithTimeout` `ctxWithCancel`.